tests/views/PageSearchPortal.spec.js

```diff
@@ -0,0 +1,149 @@
+                $router: {
+                    push: vi.fn(),
+                    resolve: () => ({ href: '/srp' }),
+                },
+                $route: {
+                    path: '/search-portal',
+                    fullPath: '/search-portal',
+                    query: {},
+                },
+                $buefy: {
+                    dialog: { alert: vi.fn() },
+                    toast: { open: vi.fn() },
+                },
+                ...mocks,
+            },
+        },
+    });
+
+describe('PageSearchPortal.vue', () => {
+    let wrapper;
+
+    afterEach(() => {
+        wrapper?.unmount();
+        vi.clearAllMocks();
+    });
+
+    it('renders search portal root', () => {
+        const store = createStoreMock();
+        wrapper = mountPage(store);
+        expect(wrapper.find('.search-portal-root').exists()).toBe(true);
+    });
+
+    it('sanitizeMobile works correctly', () => {
+        const store = createStoreMock();
+        wrapper = mountPage(store);
+        expect(wrapper.vm.sanitizeMobile('+91-9876543210')).toBe('9876543210');
+        expect(wrapper.vm.sanitizeMobile('123')).toBe(null);
+    });
+
+    it('navigates for valid mobile number', () => {
+        const push = vi.fn();
+        const store = createStoreMock();
+        wrapper = mountPage(store, { $router: { push } });
+        wrapper.vm.searchRequestWithMobile('9876543210');
+        expect(push).toHaveBeenCalled();
+    });
+
+    it('shows alert for invalid mobile number', () => {
+        const alert = vi.fn();
+        const store = createStoreMock();
+        wrapper = mountPage(store, {
+            $buefy: { dialog: { alert } },
+        });
+
+        wrapper.vm.searchRequestWithMobile('123');
+        expect(alert).toHaveBeenCalled();
+    });
+
+    it('shows alert when hasError changes to true', async () => {
+        const alert = vi.fn();
+        const store = createStoreMock();
+        wrapper = mountPage(store, {
+            $buefy: { dialog: { alert } },
+        });
+
+        store.state.searchPortal.errorMessage = 'Some error';
+        store.state.searchPortal.hasError = true;
+        await wrapper.vm.$nextTick();
+        expect(alert).toHaveBeenCalled();
+    });
+
+    it('updates route when activeTabView changes', async () => {
+        const push = vi.fn();
+        const store = createStoreMock();
+        wrapper = mountPage(store, { $router: { push } });
+        wrapper.vm.activeTabView = 1;
+        await wrapper.vm.$nextTick();
+        expect(push).toHaveBeenCalled();
+    });
+
+    it('handleConfirm navigates back', () => {
+        const push = vi.fn();
+        const store = createStoreMock({
+            searchPortal: {
+                searchMobile: '9876543210',
+                SOLatLngInput: '12,77',
+            },
+        });
+
+        wrapper = mountPage(store, { $router: { push } });
+        wrapper.vm.handleConfirm();
+        expect(push).toHaveBeenCalled();
+    });
+});
```

Comment:
The PageSearchPortal.spec.js has no toMatchSnapshot() call, but there's a snapshot file delete them if not needed.